ckan/logic/action/get.py

```diff
@@ -521,7 +521,8 @@ def organization_list(context, data_dict):
     '''
     _check_access('organization_list', context, data_dict)
     data_dict['groups'] = data_dict.pop('organizations', [])
-    data_dict['type'] = 'organization'
+    if 'type' not in data_dict:
+        data_dict['type'] = 'organization'
```

Comment:
There's [`dict.setdefault`](https://docs.python.org/2/library/stdtypes.html#dict.setdefault) for this pattern.